enhance(server): Improve user block

syuilo opened the pull request:
<!-- ℹ お読みください
PRありがとうございます！ PRを作成する前に、以下をご確認ください:
- 可能であればタイトルに、以下で示すようなPRの種類が分かるキーワードをプリフィクスしてください。
  - fix / refactor / feat / enhance / perf / chore
  - また、PRの粒度が適切であることを確認してください。ひとつのPRに複数の種類の変更や関心を含めることは避けてください。
- このPRによって解決されるIssueがある場合は、そのIssueへの参照を本文内に含めてください。
- CHANGELOG.mdに変更点を追記してください。リファクタリングなど、利用者に影響を与えない変更についてはこの限りではありません。
- この変更により新たに作成、もしくは更新すべきドキュメントがないか確認してください。
- 機能追加やバグ修正をした場合は、可能であればテストケースを追加してください。
- テスト、Lintが通っていることを予め確認してください。
  - `npm run test`、`npm run lint`でぞれぞれ実施可能です
- UIに変更がある場合はスクリーンショットを本文内に添付してください。
ご協力ありがとうございます🤗
-->
<!-- ℹ README
Thank you for your PR! Before creating a PR, please check the following:
- If possible, prefix the title with a keyword that identifies the type of this PR, as shown below.
  - fix / refactor / feat / enhance / perf / chore
  - Also, make sure that the granularity of this PR is appropriate. Please do not include more than one type of change or interest in a single PR.
- If there is an Issue which will be resolved by this PR, please include a reference to the Issue in the text.
- Please add the summary of the changes to CHANGELOG.md. However, this is not necessary for changes that do not affect the users, such as refactoring.
- Check if there are any documents that need to be created or updated due to this change.
- If you have added a feature or fixed a bug, please add a test case if possible.
- Please make sure that tests and Lint are passed in advance.
  - You can run it with `npm run test` and `npm run lint`.
- If this PR includes UI changes, please attach a screenshot in the text.
Thanks for your cooperation 🤗
-->

# What
<!-- このPRで何をしたのか？ どう変わるのか？ -->
<!-- What did you do with this PR? How will it change things? -->
ブロックの挙動を改修し、ブロックされたユーザーがブロックしたユーザーに対して干渉できないようにした

Resolve #5387
Resolve #4690 

# Why
<!-- なぜそうするのか？ どういう意図なのか？ 何が困っているのか？ -->
<!-- Why do you do it? What are your intentions? What is the problem? -->
現行のブロックはフォローをできなくする機能しかなく、直感的ではないとの意見が多かったため

# Additional info (optional)
<!-- テスト観点など -->
<!-- Test perspective, etc -->
- [x] テスト書いた
- [x] 動作確認した(テストで)
- [x] CHANGELOG書いた
- [x] ドキュメント更新した
- [x] UIに変更は無し

syuilo committed:
enhance(server): Improve user block

syuilo committed:
Update CHANGELOG.md

tamaina commented:
Issueは #4690 とか #5387 とかがあった

syuilo commented:
:azasu:

syuilo commented:
リストについて忘れてたからやる

syuilo commented:
じゃあ入れられなくするだけでなく既に入ってたら抜かす処理も要るな

syuilo committed:
Merge branch 'develop' into improve-block

mei23 commented:
TLで相手から見られなくなる、TLで相手が見れなくなる はないの？これが一番重要そうに思う

mei23 commented:
> TLで相手から見られなくなる、TLで相手が見れなくなる はないの？これが一番重要そうに思う

Issueがありそうでない

syuilo commented:
> TLで相手から見られなくなる

フォローが解除になって再度フォローすることもできなくなるからそれで事足りるかなと思った
仮にLTLに流れるような投稿も見れなくしたとしても、ログアウトとかアカウント切り替えれば普通に見れるしあんまり意味ないかと

syuilo commented:
TLで相手が見れなくなる はミュート使ってもらうとか(ブロックした際にミュートも促すようなダイアログ表示しようと思ってる)

tamaina commented:
> TLで相手から見られなくなる

- LTLでも見えなくなると文脈がわからなくなって逆に気まずくなって状況が悪化しそう
- フォロワー向け投稿は見ることができなくなるので意味はある

syuilo commented:
> > TLで相手から見られなくなる
> 
> フォローが解除になって再度フォローすることもできなくなるからそれで事足りるかなと思った
> 仮にLTLに流れるような投稿も見れなくしたとしても、ログアウトとかアカウント切り替えれば普通に見れるしあんまり意味ないかと

実装はそんなに難しくなさそうだからやっても良いとは思う

mei23 commented:
> 仮にLTLに流れるような投稿も見れなくしたとしても、ログアウトとかアカウント切り替えれば普通に見れるしあんまり意味ないかと

容易に見れないだけでも効果はあるので、見れないに越したことはないかなと。

syuilo commented:
じゃあやることにする

syuilo committed:
ユーザーリスト対応

syuilo commented:
あとアンテナにも捕捉されないようにするか

syuilo committed:
相手から見れなくなるように

syuilo commented:
> じゃあやることにする

やった https://github.com/misskey-dev/misskey/pull/7640/commits/b690c410a37b129e42fb35d759dd0dedc2dca1f7

rinsuki commented on test/block.ts:
```diff
@@ -0,0 +1,74 @@
+	it('ブロックされているユーザーをフォローできない', async(async () => {
+		const res = await request('/following/create', { userId: alice.id }, bob);
+
+		assert.strictEqual(res.status, 400); // TODO: エラーID確認
```
:eyes:

syuilo replied:
updated

rinsuki reviewed (commented):
ブロックしているユーザーにも干渉できないようにすべきかも (Mastodonはそうなっている？)

syuilo committed:
Update 1629004542760-chart-reindex.ts

https://github.com/misskey-dev/misskey/commit/2365761ba5445f26c8b66b3b20ef4be44e70d549#commitcomment-54919821

syuilo committed:
Merge branch 'develop' into improve-block

syuilo committed:
update test

syuilo committed:
add test

syuilo committed:
add todos

syuilo committed:
Update 1629004542760-chart-reindex.ts

syuilo commented:
> ブロックしているユーザーにも干渉できないようにすべきかも (Mastodonはそうなっている？)

ブロックされている場合としている場合とでエラーメッセージ出し分けしないといけなくて実装増えそうだからこのPRでは対応しないことにする

syuilo commented:
マージするか
